src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -409,6 +411,17 @@ const buildGeneratorLoop: LoopBuilder = makeForLoopBuilder((state, initializer,
 });
 
 function getLoopBuilder(state: TransformState, node: ts.Node, type: ts.Type): LoopBuilder {
+	// Check for 'any' or 'unknown' types first to provide better error messages
+	if (isDefinitelyType(type, isAnyType(state))) {
+		DiagnosticService.addDiagnostic(errors.noAnyForOfIteration(node));
+		return () => luau.list.make();
+	}
+
+	if (isDefinitelyType(type, isUnknownType())) {
+		DiagnosticService.addDiagnostic(errors.noAnyForOfIteration(node));
+		return () => luau.list.make();
+	}
```

Comment:
Typescript already disallows iterating over `unknown`, so this check is unnecessary.